backend/src/routes/Passwordless.ts

```diff
@@ -0,0 +1,109 @@
+import express, { Router } from 'express';
+import * as z from 'zod';
+import type { User, Auth0, PasswordlessRequest } from '../services';
+import { normalizePhoneNumber } from '../util/country';
+import { validateRequest } from './helpers/validation';
+
+const PASSWORDLESS_REQUEST_INTERVAL_MILLIS = 1000 * 60 * 2;
+
+const BEGIN_SCHEMA = z.object({
+  body: z.object({
+    phoneNumber: z.string(),
+  }),
+});
+
+const LOGIN_SCHEMA = z.object({
+  body: z.object({
+    phoneNumber: z.string(),
+    code: z.string(),
+  }),
+});
+
+function PasswordlessRoutes(
+  userService: typeof User,
+  auth0Service: typeof Auth0,
+  passwordlessRequestService: typeof PasswordlessRequest
+): Router {
+  const router = express.Router();
+
+  router.post(
+    '/begin',
+    validateRequest(BEGIN_SCHEMA, async (parsedReq, res, req) => {
+      const { id } = req.user;
```

Comment:
I think we should add `req.user` this to the schema, what do you think?

Reply:
Seems like different things though. the schema is meant for stuff coming from the client, whereas user is coming from the middleware. like someone might expect user.id to be somewhere in the request sent from the browser if they didnt know.

Reply:
I see, I think we're looking at this two different ways. You're thinking of using the schema to guard against client behavior and I'm thinking of using the schema to give type safety to route handlers. I'm ok with sticking with your approach, it sounds reasonable. I will add zod validation to the middleware too so that we can have safety for `req.user`.